test/tests_unittests/addons/addon.js

```diff
@@ -0,0 +1,326 @@
+/* global describe beforeEach it */
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const logger = require('winston');
+const path = require('path');
+
+logger.level = 'error';
+
+const expect = chai.expect;
+chai.use(chaiAsPromised);
+
+const cachePath = path.resolve('./app/addons/addon.js');
+
+let Addon;
+let addon;
+let addonPrototype;
+
+describe('addon.js', function () {
+  beforeEach(function (done) {
+    delete require.cache[cachePath];
+    Addon = require('../../../app/addons/addon'); // eslint-disable-line
+
+    addon = new Addon('naim', true);
+    addon.addonPath = path.join(__dirname, 'mocking');
+
+    addonPrototype = Object.getPrototypeOf(addon);
+    done();
+  });
+
+  describe('toJson', function () {
+    it('toJson - uncorrupted addon', function (done) {
+      addon.description = 'description data';
+      addon.version = 'version data';
+      addon.repository = 'repository of addon';
+      addon.hasStaticContent = false;
+
+      const jsonData = addon.toJson;
+      expect(jsonData).to.have.property('description', 'description data');
+      expect(jsonData).to.have.property('version', 'version data');
+      expect(jsonData).to.have.property('repository', 'repository of addon');
+      expect(jsonData).to.have.property('status', 'introduce-done');
+      expect(jsonData).to.have.property('hasStaticContent', false);
+
+      done();
+    });
+  });
+
+  describe('Status', function () {
+    it('Status - valid state', function (done) {
+      addon._status = { state: 1, phase: 2 };
+      expect(addon.Status).to.equal('load-failed');
+      done();
+    });
+  });
+
+  describe('isBusy', function () {
+    it('isBusy - true and false case', function (done) {
+      addon._status = { state: 0, phase: 1 };
```

Comment:
plain numbers are not very describe for this kind of thing, some freezed object would be better..